core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataClient.java

```diff
@@ -46,12 +49,9 @@
 @SdkInternalApi
 @Immutable
 @ThreadSafe
-public final class DefaultEc2Metadata implements Ec2Metadata {
-
-    private static final Logger log = Logger.loggerFor(DefaultEc2Metadata.class);
+public final class DefaultEc2MetadataClient implements Ec2MetadataClient, SdkClient {
```

Comment:
Shouldn't it?

Reply:
We've viewed it as a kind of utility that isn't a regular SDK generated client. Since this is the first of its kind it makes sense to discuss this further and the pros/cons of each approach. We can add this topic to a surface API review list and not change it in this PR.